influence_benchmark/experiments/expert_iteration_test.py

```diff
@@ -0,0 +1,102 @@
+import multiprocessing as mp
+
+import torch
+
+from influence_benchmark.RL.expert_iteration import ExpertIteration
+from influence_benchmark.root import PROJECT_ROOT
+from influence_benchmark.utils.utils import set_all_seeds
+
+DEBUG = False
+
+if __name__ == "__main__":
+    mp.set_start_method("spawn", force=True)
+
+if DEBUG:
+    # Debugging CUDA devices
+    if torch.cuda.is_available():
+        print(f"Available CUDA devices: {torch.cuda.device_count()}")
+        for i in range(torch.cuda.device_count()):
+            print(f"Device {i}: {torch.cuda.get_device_name(i)}")
+    else:
+        print("CUDA is not available.")
+
+
+def main():
+    # Seed everything
+    seed = 42
+    set_all_seeds(seed)
+
+    # Specify settings for generating trajectories
+    env_name = "n_test"
+    # number of back and forths in each conversation
+    max_turns = 2
+    # number of environment slots to be filled with env-subenv-initialstate combinations. # 8 is roughly max
+    num_envs_per_device = 8
+    num_gen_trajs_per_initial_state = 16
+    top_n_trajs_per_initial_state = 1  # on a single GPU across all trajactories
+    iterations = 2
+    ignore_first_n_assistant_messages = 1  # Number of assistant messages to not train on
+    run_name = None
+    # GPUs used for generating trajectories. The GPUs used for training are specified in the accelerate_config.yaml file.
+    devices = [2, 3]
+    mode = "multi"  # running on multiple environemnts in parallel
+    log_to_wandb = True
+    assert num_gen_trajs_per_initial_state >= top_n_trajs_per_initial_state
+
+    env_args = {
+        "env_name": env_name,
+        "max_turns": max_turns,
+        "print": False,
+        "num_envs_per_device": num_envs_per_device,
+        "vectorized": True,
+    }
+
+    # Specify settings for training
+    agent_model_name = "meta-llama/Meta-Llama-3-8B-Instruct"
+    env_model_name = "meta-llama/Meta-Llama-3-8B-Instruct"
+    accelerate_config_path = str(PROJECT_ROOT / "RL" / "accelerate_slurm.yaml")
+    script_path = str(PROJECT_ROOT / "RL" / "SFT.py")
+
+    training_args = {
+        "agent_model_name": agent_model_name,
+        "env_model_name": env_model_name,
+        "per_device_train_batch_size": 1,
+        "num_train_epochs": 1,
+        "gradient_accumulation_steps": 32,  # Number of steps to accumulate gradients before performing an update.
```

Comment:
32 is a bit high especially for test this means there will only be 1 gradient step?